// eslint-config-settled/index.js
/*
 * ESLint config used by Settled.co.uk. Based on AirBnB's implementation.
 */

module.exports = {

  extends: 'airbnb',

  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'script',
    ecmaFeatures: {
      impliedStrict: false,
      jsx: true
    }
  },

  env: {
    'shared-node-browser': true
  },

  rules: {
    'arrow-body-style': [1, 'as-needed'],
    'arrow-parens': 0,
    'brace-style': [2, '1tbs', {
      'allowSingleLine': true
    }],
    'callback-return': [2, ['callback', 'cb', 'finish', 'done', 'next', 'nextItem']],
    'camelcase': 1,
    'consistent-return': 0,
    'curly': [2, 'all'],
    'func-names': 0,
    'global-require': 1,
    'guard-for-in': 2,
    'handle-callback-err': [2, "^err"],
    "id-length": [1, {
      'min': 2,
      'max': 35
    }],
    'max-depth': [2, 10],
    'max-len': [1, 120],
    'max-nested-callbacks': [2, 5],
    'max-params': [2, 5],
    'new-cap': [2, {
      'newIsCap': true,
      'capIsNew': true,
      'capIsNewExceptions': ['Router']
    }],
    'newline-after-var': [2, 'always'],
    'no-cond-assign': [2, 'except-parens'],
    'no-confusing-arrow': 2,
    'no-console': 1,
    'no-continue': 0,
    'no-else-return': 0,
    'no-lonely-if': 2,
    'no-mixed-requires': [1, {
      'grouping': true,
      'allowCall': true
    }],
    'no-new-require': 2,
    'no-param-reassign': [2, {
      'props': false
    }],
    'no-path-concat': 2,
    'no-plusplus': [1, {
      'allowForLoopAfterthoughts': true
    }],
    'no-process-exit': 2,
    'no-return-assign': 0,
    'no-shadow': [2, {
      'allow': ['err']
    }],
    'no-sync': 2,
    'no-underscore-dangle': 0,
    'no-unexpected-multiline': 2,
    'no-unused-expressions': [2, {
      'allowTernary': true
    }],
    'no-unused-vars': [2, {
      'vars': 'all',
      'args': 'after-used'
    }],
    'no-with': 2,
    'object-curly-spacing': [2, 'always'],
    'one-var': [2, 'never'],
    'padded-blocks': 0,
    'prefer-arrow-callback': 1,
    'prefer-const': 2,
    'prefer-spread': 2,
    'radix': 0,
    'semi': [2, 'always'],
    'space-before-blocks': [1, {
      'functions': 'always',
      'keywords': 'always',
      'classes': 'always'
    }],
    'space-before-function-paren': [1, {
      'anonymous': 'always',
      'named': 'always'
    }],
    'spaced-comment': 1,
    'strict': [2, 'global'],
    'yoda': [2, 'never']
  }

};
